src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,223 @@
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carsIndexes.add(resultSet.getObject("id", Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars from DB", e);
+        }
+        return carsIndexes.stream()
+                .map(i -> get(i).orElseThrow(() ->
+                        new RuntimeException("Can't get car from DB by id: " + i)))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ?"
+                + " WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionModule.getConnection();
+                 PreparedStatement statement
+                         = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car " + car, e);
+        }
+        refreshDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionModule.getConnection();
+                 PreparedStatement statement
+                         = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car from DB by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT id"
+                + " FROM cars c"
+                + " JOIN cars_drivers cd"
+                + " ON c.id = cd.driver_id"
+                + " WHERE cd.car_id = ? AND c.is_deleted = FALSE;";
+        List<Long> carIndexes = new ArrayList<>();
+        try (Connection connection = ConnectionModule.getConnection();
+                 PreparedStatement statement
+                         = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carIndexes.add(resultSet.getObject("id", Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get cars from DB by driver id: "
+                    + driverId, e);
+        }
+        return carIndexes.stream()
+                .map(i -> get(i).orElseThrow(() ->
+                        new DataProcessingException("Can't get car from DB by id: " + i, null)))
```

Comment:
you can use constructor with one param

Reply:
no i can't because
if i remove second argument i getting compiler error

Reply:
no i can't because
if i remove second argument i getting compiler error

Reply:
ok